### PR TITLE
feat: filter transfers of omnipool assets to omnipool

### PR DESCRIPTION
**Pull request by Roznovjak**
Fixes: #565

### CODE REVIEW

File: runtime/hydradx/src/lib.rs [hunk 1]
```diff
@@ -207,6 +207,7 @@ impl Contains<Call> for CallFilter {
 			// Lookup::lookup() is not necessary thanks to IdentityLookup
 			if dest == &Omnipool::protocol_account()
 				&& *currency_id == <Runtime as pallet_omnipool::Config>::HubAssetId::get()
+				|| Omnipool::exists(*currency_id)
```
Comment (enthusiastmartin) [inline runtime/hydradx/src/lib.rs:210]: this filters all transfers to any account if asset is in the omnipool.

**Review by enthusiastmartin — CHANGES_REQUESTED** [review]

File: runtime/hydradx/src/lib.rs (hunk 1 above)
Comment (enthusiastmartin) [reply runtime/hydradx/src/lib.rs:210]: this also allows HDX to be transfered. We need to filter out balances transfer too.

Comment (Roznovjak) [reply runtime/hydradx/src/lib.rs]: Nice catch, thanks. Fixed.

File: integration-tests/src/non_native_fee.rs [hunk 2]
```diff
@@ -65,22 +63,10 @@ fn non_native_fee_payment_works_with_omnipool_spot_price() {
 			0,
 		));
 
-		let native_price = FixedU128::from_inner(1201500000000000);
-		let stable_price = FixedU128::from_inner(45_000_000_000);
-		hydradx_runtime::Omnipool::protocol_account();
+		init_omnipool();
+		//let spot_price = hydradx_runtime::Omnipool::spot_price(HDX, DAI);
```
Comment (enthusiastmartin) [inline integration-tests/src/non_native_fee.rs:67]: you can remove this commented code.

**Review by enthusiastmartin — APPROVED** [review]

**Pull request merged**